x/participation/keeper/total_allocation_test.go

```diff
@@ -0,0 +1,47 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+func createNDelegations(ctx sdk.Context, tk testkeeper.TestKeepers, addr string, n int) ([]stakingtypes.Delegation, sdk.Dec) {
+	items := make([]stakingtypes.Delegation, n)
+
+	totalShares := sdk.ZeroDec()
+	for i := range items {
+		items[i] = sample.Delegation(tk.T, addr)
+		totalShares = totalShares.Add(items[i].Shares)
+		tk.StakingKeeper.SetDelegation(ctx, items[i])
+	}
+
+	return items, totalShares
+}
+
+func TestTotalAllcationGet(t *testing.T) {
```

Comment:
I think the test would be more maintainable if we explicitly define the amount to be delegated and hardcode the expected total allocation. We should not add the logic of the total allocation computation in the test

Table driven test can be great also I think if we extend this method in the future